test/integration/change-streams/change_streams.prose.test.js

```diff
@@ -0,0 +1,737 @@
+const { expect } = require('chai');
+const { LEGACY_HELLO_COMMAND } = require('../../../src/constants');
+
+const { setupDatabase } = require('../shared');
+const mock = require('../../tools/mongodb-mock/index');
+
+const sinon = require('sinon');
+const { ObjectId, Timestamp, Long, MongoNetworkError } = require('../../../src');
+
+/**
+ * Triggers a fake resumable error on a change stream
+ *
+ * @param {ChangeStream} changeStream
+ * @param {number} [delay] optional delay before triggering error
+ * @param {Function} onClose callback when cursor closed due this error
+ */
+function triggerResumableError(changeStream, delay, onClose) {
+  if (arguments.length === 2) {
+    onClose = delay;
+    delay = undefined;
+  }
+
+  const stub = sinon.stub(changeStream.cursor, 'close');
+  stub.callsFake(function () {
+    stub.wrappedMethod.call(this);
+    stub.restore();
+    onClose();
+  });
+
+  function triggerError() {
+    const cursorStream = changeStream.cursorStream;
+    if (cursorStream) {
+      cursorStream.emit('error', new MongoNetworkError('error triggered from test'));
+      return;
+    }
+
+    const nextStub = sinon.stub(changeStream.cursor, 'next').callsFake(function (callback) {
+      callback(new MongoNetworkError('error triggered from test'));
+      nextStub.restore();
+    });
+
+    changeStream.next(() => {});
+  }
+
+  if (delay != null) {
+    setTimeout(triggerError, delay);
+    return;
+  }
+
+  triggerError();
+}
+
+/**
+ * Waits for a change stream to start
+ *
+ * @param {ChangeStream} changeStream
+ * @param {Function} callback
+ */
+function waitForStarted(changeStream, callback) {
+  changeStream.cursor.once('init', () => {
+    callback();
+  });
+}
+
+describe('Change Stream prose tests', function () {
+  before(async function () {
+    return await setupDatabase(this.configuration, ['integration_tests']);
+  });
+
+  beforeEach(async function () {
+    const configuration = this.configuration;
+    const client = configuration.newClient();
+
+    await client.connect();
+    const db = client.db('integration_tests');
+    try {
+      await db.createCollection('test');
+    } catch {
+      // ns already exists, don't care
+    } finally {
+      await client.close();
+    }
+  });
+  afterEach(async () => await mock.cleanup());
+
+  // TODO(NODE-3884): Add tests 1-4, 6-9.
```

Comment:
Can you mention 5 removed here, just saves a lookup

Reply:
sure